packages/react/src/side-nav/SideNav.test.tsx

```diff
@@ -66,8 +66,8 @@ describe('SideNav', () => {
 			});
 
 			const nav = screen.getByRole('navigation', { hidden: true });
-			const el = within(nav).getByText(defaultTestingProps.title);
-			expect(el).toBeInTheDocument();
+			const els = within(nav).getAllByText(defaultTestingProps.title);
+			expect(els.length).toEqual(3);
```

Comment:
@ChrisLaneAU There are now 3 spans with the same text. This doesn't feel like a good test though. The one I want is marked with ******:

```
<nav
      aria-labelledby="collapsing-side-bar-:r4:-heading"
      class="css-2yhndx-boxStyles-CollapsingSideBar"
    >
      <div
        class="css-bmr02n-boxStyles-CollapsingSideBar"
      >
        <div
          class="css-lr9d04-boxStyles-CollapsingSideBar"
          id="collapsing-side-bar-:r4:-heading"
        >
          <h2
            class="css-m2oixi-boxStyles-SideNav"
          >
            <span
              class="css-1jga52b-boxStyles"
            >
              Lodging your tax return
            </span>
          </h2>
        </div>
        <button
          aria-controls="collapsing-side-bar-:r4:-body"
          aria-expanded="false"
          aria-label="Lodging your tax return"
          class="css-1c4meu7-BaseButton-boxStyles-CollapsingSideBar"
          type="button"
        >
          <span
            class="css-1eiy0fa-boxStyles"
          >
            <span
              class="css-1wcx0lf-boxStyles"
            >
              <span
                class="css-1rvfmyi-boxStyles-CollapsingSideBar"
              >
                Lodging your tax return
              </span>
            </span>
            <svg
              aria-hidden="true"
              class="css-ln9yj9-CollapsingSideBar"
              clip-rule="evenodd"
              fill-rule="evenodd"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="css-3b4a55-CollapsingSideBar"
        id="collapsing-side-bar-:r4:-body"
        style="display: none; height: 0px;"
      >
        <div
          class="css-79i25n-boxStyles"
        >
          <div
            class="css-zipzg2-boxStyles"
          >
            <span
              class="css-1lnm2ec-boxStyles"
            >
              <span
                class="css-1jga52b-boxStyles"
              >
                ******Lodging your tax return
              </span>
            </span>
            <ul
              class="css-79i25n-boxStyles"
            >
              <li
                class="css-0"
              >
                <a
                  class="css-1xapbrs-boxStyles-SideNavLink"
                  href="/welcome"
                >
                  <span
                    class="css-127d72z-SideNavLink"
                  >
                    Welcome
                  </span>
                </a>
              </li>
```

Reply:
Yeah, this is a little tricky because you can't test for role="generic". What you can do is extend your test to assert that there is no link with the accessible name of `Lodging your tax return`. This way you assert that there are still three elements with the title text but none of them are a link. You could do the converse in another test with a `titleLink` where you assert that there is an element with role="link" and that title text is the accessible name.

So I'd split the tests into four within a describe, i.e. one for accessibility violations, one for html violations, one for the absence of a role="link", and one for having three elements with the text. _Note: I remember updating this test a little while ago and struggling with the async expanding of the component but that seems fine locally for me now_:

```ts
		describe('when there is no `titleLink` defined', () => {
			test('then there are no accessibility violations', async () => {
				const { container } = renderSideNav({
					...defaultTestingProps,
					titleLink: undefined,
				});

				expect(await axe(container)).toHaveNoViolations();
			});

			test('then there are no html violations', async () => {
				const { container } = renderSideNav({
					...defaultTestingProps,
					titleLink: undefined,
				});

				expect(container).toHTMLValidate({
					extends: ['html-validate:recommended'],
					rules: {
						'no-inline-style': 'off',
						// react 18s `useId` break this rule
						'valid-id': 'off',
					},
				});
			});

			test('then there is no link with that title text', async () => {
				renderSideNav({
					...defaultTestingProps,
					titleLink: undefined,
				});

				const user = userEvent.setup();
				const button = screen.getByRole('button', {
					name: defaultTestingProps.title,
				});
				await user.click(button);

				expect(
					screen.queryByRole('link', { name: defaultTestingProps.title })
				).toBeNull();
			});

			test('then there are three elements with that title text', async () => {
				renderSideNav({
					...defaultTestingProps,
					titleLink: undefined,
				});

				const user = userEvent.setup();
				const button = screen.getByRole('button', {
					name: defaultTestingProps.title,
				});
				await user.click(button);

				const elementsWithTitleText = await screen.findAllByText(
					defaultTestingProps.title
				);

				expect(elementsWithTitleText).toHaveLength(3);
			});
		});
```

Then that converse test when the `titleLink` is defined would be something like:

```ts
			test('then the title text is rendered as a link', async () => {
				renderSideNav(defaultTestingProps);

				const user = userEvent.setup();
				const button = screen.getByRole('button', {
					name: defaultTestingProps.title,
				});
				await user.click(button);

				const titleAsLink = await screen.findByRole('link', {
					name: defaultTestingProps.title,
				});

				expect(titleAsLink).toBeVisible();
			});
```

Reply:
Thanks, my test wasn't actually doing the right thing. I have just updated with a variation of yours (but being a bit more consistent with the rest)